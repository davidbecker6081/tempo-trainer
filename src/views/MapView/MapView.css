.map-container {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 4em;
  padding: 2rem;
  width: 60vw;
  background-color: #72C6E9;
}

.google-map-wrapper {
  height: 30vh;
  width: 45vw;
  border: 2px solid #FFFFFF;
  border-radius: 3px;
}
