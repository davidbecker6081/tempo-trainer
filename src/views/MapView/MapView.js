import React from 'react';
import PropTypes from 'prop-types';
import GoogleApiWrapper from '../../components/Map/GoogleMap';
import './MapView.css';

const MapView = ({ range, dataHelper, originalRange }) =>
  (
    <div>
      <GoogleApiWrapper
        className="google-map-wrapper"
        range={range}
        dataHelper={dataHelper}
        originalRange={originalRange}
      />
    </div>
  );

export default MapView;

MapView.propTypes = {
  dataHelper: PropTypes.shape({
    data: PropTypes.shape({}),
    channels: PropTypes.arrayOf(PropTypes.string),
    GPSCoords: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
  originalRange: PropTypes.arrayOf(PropTypes.number).isRequired,
  range: PropTypes.arrayOf(PropTypes.number).isRequired,
};
