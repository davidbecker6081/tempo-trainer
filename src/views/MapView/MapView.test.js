import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import MapView from './MapView';
import DataHelper from '../../components/DataCleaner/DataCleaner';
import workoutData from '../../__mock__/workout-data.json';

describe('MapView', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      originalRange: [],
      dataHelper: {},
      range: [],
    };
  });

  describe('originalRange is defined', () => {
    beforeEach(() => { mockProps.originalRange = [1, 5]; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('range is defined', () => {
    beforeEach(() => { mockProps.range = [0, 5]; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.dataHelper = new DataHelper(workoutData); });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
