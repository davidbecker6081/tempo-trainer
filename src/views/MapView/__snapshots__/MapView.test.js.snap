// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapView dataHelper is defined should render correclty 1`] = `undefined`;

exports[`MapView originalRange is defined should render correclty 1`] = `undefined`;

exports[`MapView range is defined should render correclty 1`] = `undefined`;
